worlds/blasphemous/Rules.py

```diff
@@ -0,0 +1,1505 @@
+                            state._blasphemous_blood_relic(player))
+    set_rule(world.get_location("Confessor Dungeon 7 main", player),
+        lambda state: state._blasphemous_bead(player) and \
+            state._blasphemous_bridge_access(player) and \
+                state._blasphemous_1_mask(player) and \
+                    state._blasphemous_bronze_key(player) and \
+                        state._blasphemous_silver_key(player) and \
+                            state._blasphemous_blood_relic(player))
+    # to do: requires dive
+    set_rule(world.get_location("Defeat 1 Amanecida", player),
+        lambda state: state._blasphemous_bell(player) and \
+            state._blasphemous_open_holes(player))
+    set_rule(world.get_location("Defeat 2 Amanecidas", player),
+        lambda state: state._blasphemous_bell(player) and \
+            state._blasphemous_open_holes(player) and \
+                state._blasphemous_blood_relic(player) and \
+                    (state._blasphemous_root_relic(player) or \
+                        state._blasphemous_bridge_access(player)))
+    set_rule(world.get_location("Defeat 3 Amanecidas", player),
+        lambda state: state._blasphemous_bell(player) and \
+            state._blasphemous_open_holes(player) and \
+                state._blasphemous_bridge_access(player) and \
+                    state._blasphemous_blood_relic(player) and \
+                        (state._blasphemous_root_relic(player) or \
+                            (state._blasphemous_1_mask(player) and \
+                                state._blasphemous_bronze_key(player) and \
+                                    state._blasphemous_silver_key(player))))
+    set_rule(world.get_location("Defeat 4 Amanecidas", player),
+        lambda state: state._blasphemous_bell(player) and \
+            state._blasphemous_open_holes(player) and \
+                state._blasphemous_bridge_access(player) and \
+                    state._blasphemous_1_mask(player) and \
+                        state._blasphemous_blood_relic(player) and \
+                            state._blasphemous_root_relic(player) and \
+                                state._blasphemous_bronze_key(player) and \
+                                    state._blasphemous_silver_key(player))
+    set_rule(world.get_location("Defeat all Amanecidas", player),
+        lambda state: state._blasphemous_bell(player) and \
+            state._blasphemous_open_holes(player) and \
+                state._blasphemous_bridge_access(player) and \
+                    state._blasphemous_1_mask(player) and \
+                        state._blasphemous_blood_relic(player) and \
+                            state._blasphemous_root_relic(player) and \
+                                state._blasphemous_bronze_key(player) and \
+                                    state._blasphemous_silver_key(player))
+
+    # expert logic
+    if world.expert_logic[player]:
```

Comment:
If I always set the expert rules and then added on top of them, wouldn't that effectively do nothing? If the expert rules are always there, wouldn't the randomizer always have a chance of using the more difficult solution, even with the option turned off?

Reply:
`add_rule` defaults to `and`, with `or` being optional so it would be `old_rule and new_rule`. alternatively `set_rule` would just override it, but in lots of cases this is something where you want and because it might be "have this item and this other item" to reach it.

Reply:
Oh, now I get what you're saying. But what I already had was already working anyways, so unless it would be faster if I rewrote them using `add_rule` instead of `set_rule` then I don't see why it needs to be changed.